src/components/helpers/pretty-text-input.js

```diff
@@ -179,14 +179,7 @@ export default createReactClass({
   },
 
   onBlur() {
-    if (this.isDebouncingCodeMirrorChange) {
-      this.onChangeAndTagCodeMirror();
-      this.isDebouncingCodeMirrorChange = false;
-      this.setState({ hasFocus: false }, this.props.onBlur);
-    } else {
-      this.setState({ hasFocus: false });
-      this.props.onBlur();
-    }
+    this.setState({ hasFocus: false }, this.props.onBlur);
```

Comment:
I assumed that in both case we can invoke `this.props.onBlur` using the `Component#setState` callback, it helps to simplify the code a little bit.

Another assumption I made is that we don't necessarily need to invoke `onChangeAndTagCodeMirror` here as it will be invoked shortly after the `blur` event and I don't see any outstanding issue of doing so, but I'm not 💯 sure of that.

Reply:
Yeah, always calling `onBlur` in `setState` seems fine. Hopefully there are no dragons there.

I _think_ the other change is problematic though, and it's related to the bug you're fixing here. If you blur away and changes are being debounced, if the field gets unmounted before the debounce, we'll lose the changes. This commit seems to be intentionally fixing something, and I think that's what it was, since I remember that problem happening: https://github.com/zapier/formatic/commit/11ebdb58e1bd42d2b58c33f94fa2e06d0fbc9d00.

Reply:
Oh gotcha! So if I understand correctly we could hit a edge case where the component is umounted within the 200ms timespan after the user blurs the field. If so we effectively lose the last batch of input...